code/game/objects/structures/tables_racks.dm

```diff
@@ -263,29 +263,54 @@
 	buildstack = /obj/item/stack/sheet/glass
 	canSmoothWith = null
 	health = 50
+	var/list/debris = list()
+
+/obj/structure/table/glass/New()
+	. = ..()
+	debris += new frame
+	debris += new /obj/item/weapon/shard
+
+/obj/structure/table/glass/Destroy()
+	for(var/i in debris)
+		qdel(i)
+	. = ..()
 
 /obj/structure/table/glass/Crossed(atom/movable/AM)
 	. = ..()
 	if(flags & NODECONSTRUCT)
 		return
+	if(!isliving(AM))
+		return
 	// Don't break if they're just flying past
 	if(AM.throwing)
-		return
+		spawn(5)

```

Comment:
AM.throwing can be set to false in at least three different places.

1) When an atom Bumps another
2) When the throw hits something early
3) When the thrown item hits the turf of its target

I don't understand throw_impact enough to be able to change it. The side effects of being thrown onto a tile full of broken glass being painful is a really good emergent property, but I don't feel comfortable tackling that in this PR. Spawn will have to stay.